HLTrigger/special/src/HLTMultipletFilter.cc

```diff
@@ -0,0 +1,177 @@
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("L1EGammaInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1EtSumInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1JetInputTag",edm::InputTag("hltCaloStage2Digis:Jet"));
+  desc.add<edm::InputTag>("L1MuonInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1TauInputTag",edm::InputTag("hltCaloStage2Digis:Tau"));
+  desc.add<int>("MinN",1);
+  desc.add<double>("MinEta",1.305);
+  desc.add<double>("MaxEta",3.000);
+  desc.add<double>("MinPhi",5.4105);
+  desc.add<double>("MaxPhi",5.5796);
+  desc.add<double>("MinPt", 20.0);
+  descriptions.add("hltMultipletFilter",desc);
+}
+
+bool HLTMultipletFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSetup, trigger::TriggerFilterObjectWithRefs & filterproduct) const {
+
+  // the filter object
+  if (saveTags()) {
+    for (int type=0; type<nobj_; ++type) {
+      if (flag_[type]) {
+	switch (type) {
+	  case EGamma:
+	    filterproduct.addCollectionTag(hltEGammaSeedLabel_);
+	    break;
+	  case EtSum:
+	    filterproduct.addCollectionTag(hltEtSumSeedLabel_);
+	    break;
+	  case Jet:
+	    filterproduct.addCollectionTag(hltJetSeedLabel_);
+	    break;
+ 	  case Muon:
+	    filterproduct.addCollectionTag(hltMuonSeedLabel_);
+	    break;
+	  default:
+	    filterproduct.addCollectionTag(hltTauSeedLabel_);
+	    break;
+	}
+      }
+    }
+  }
+
+  bool      accept(false);
+  const int ibx(0);
+  int       nobj(0);
+
+  for (int type=0; type<nobj_; ++type) {
+    if (flag_[type]) {
+      switch (type) {
+        case EGamma: 	
+	  nobj += objects(iEvent,hltEGammaToken_,hltEGammaSeedLabel_,type,ibx);
+	  break;
+        case EtSum: 	
+	  nobj += objects(iEvent,hltEtSumToken_,hltEtSumSeedLabel_,type,ibx);
+	  break;
+        case Jet:
+	  nobj += objects(iEvent,hltJetToken_,hltJetSeedLabel_,type,ibx);
+	  break;
+        case Muon:
+	  nobj += objects(iEvent,hltMuonToken_,hltMuonSeedLabel_,type,ibx);
+	  break;
+        default:
+	  nobj += objects(iEvent,hltTauToken_,hltTauSeedLabel_,type,ibx);
+	  break;
+      }
+      if (nobj > minN_) accept = true;
+      if (accept) break;
+    }
+  }
+      
+  edm::LogVerbatim("Report") << "Selection flag " << accept;
+  return accept;
+}
+
+template<typename T1>
+int HLTMultipletFilter::objects(edm::Event& iEvent, 
+				edm::EDGetTokenT<T1> const& hltToken,
+				edm::InputTag const& hltSeedLabel,
+				int type, int ibx) const {
+  int nobj(0);
+  edm::Handle<T1> objs;
+  iEvent.getByToken(hltToken, objs);
+  if (!objs.isValid()) { 
+    edm::LogWarning("Report") << "Collection with input tag " << hltSeedLabel
+			      << " requested, but not found in the event.";
+  } else {
+    edm::LogVerbatim("Report") << "Collection for type " << type 
+			       << " has " << objs->size() 
+			       << " BX's and " << objs->size(ibx)
+			       << " candidates in BX " << ibx;
+    for (typename T1::const_iterator p=objs->begin(ibx); 
```

Comment:
A range based for loop - see the examples at the bottom of 
http://en.cppreference.com/w/cpp/language/range-for